lib/secretsscanner/authorizedkeys/authorized_keys.go

```diff
@@ -79,17 +78,17 @@ type WatcherConfig struct {
 	// getRuntimeOS returns the runtime operating system.
 	// used for testing purposes.
 	getRuntimeOS func() string
-	// etcPasswdFile is the path to the file that contains the users account information on the system.
-	// This file is used to get the list of users on the system and their home directories.
-	// Value is set to "/etc/passwd" by default.
-	etcPasswdFile string
+	// getHostUsers is a function that returns the list of users on the system.
+	// used for testing purposes. When nil, it uses the default implementation
+	// that leverages getpwent.
+	getHostUsers func() ([]user.User, error)
 }
 
 // NewWatcher creates a new Watcher instance.
 // Returns [ErrUnsupportedPlatform] if the operating system is not supported.
 func NewWatcher(ctx context.Context, config WatcherConfig) (*Watcher, error) {
 
-	if getOS(config) != constants.LinuxOS {
+	if platform := getOS(config); platform != constants.LinuxOS && platform != constants.DarwinOS {
```

Comment:
nit: switch with empty `linux, macos` case and error as `default` is probably more readable, but it won't make a huge different.

Reply:
Handled in [ada9362](https://github.com/gravitational/teleport/pull/45109/commits/ada936204f13cdd26392b45126d32a65a0efb5af)